lib/mime.c

```diff
@@ -279,29 +280,66 @@ static void mimesetstate(struct mime_state *state,
 
 
 /* Escape header string into allocated memory. */
-static char *escape_string(const char *src)
+static CURLcode escape_mime_string(struct dynbuf *db, const char *src)
 {
-  size_t bytecount = 0;
-  size_t i;
-  char *dst;
+  CURLcode result = CURLE_OK;
 
-  for(i = 0; src[i]; i++)
-    if(src[i] == '"' || src[i] == '\\')
-      bytecount++;
+  for(; !result && *src; result = Curl_dyn_addn(db, src++, 1))
+    if(*src == '\\' || *src == '"') {
+      result = Curl_dyn_add(db, "\\");
+      if(result)
+        break;
+    }
+  return result;
+}
 
-  bytecount += i;
-  dst = malloc(bytecount + 1);
-  if(!dst)
-    return NULL;
+static CURLcode escape_form_string(struct Curl_easy *easy,
+                                   struct dynbuf *db, const char *src)
+{
+  CURLcode result = CURLE_OK;
+
+  (void) easy;
```

Comment:
Why have the `easy` argument to the function if it's not used? Also, we call the local `struct Curl_easy *` variable `data` in 99.97% of all functions so I think sticking to that name makes sense.

Reply:
It is used if `CURL_DOES_CONVERSION` is defined.
It is called `easy` as in other functions of this module. Maybe we could change it for all the module's function.

Reply:
`esay` renamed as `data`.